mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/services/NavAnsattService.kt

```diff
@@ -82,30 +95,83 @@ class NavAnsattService(
 
         logger.info("Oppdaterer ${ansatteToUpsert.size} NavAnsatt fra Azure")
         ansatteToUpsert.forEach { ansatt ->
-            ansatte.upsert(NavAnsattDbo.fromNavAnsattDto(ansatt))
+            navAnsattRepository.upsert(NavAnsattDbo.fromNavAnsattDto(ansatt))
         }
         upsertSanityAnsatte(ansatteToUpsert)
 
         val ansatteAzureIds = ansatteToUpsert.map { it.azureId }
-        val ansatteToScheduleForDeletion = ansatte.getAll().filter { ansatt ->
+        val ansatteToScheduleForDeletion = navAnsattRepository.getAll().filter { ansatt ->
             ansatt.azureId !in ansatteAzureIds && ansatt.skalSlettesDato == null
         }
         ansatteToScheduleForDeletion.forEach { ansatt ->
             logger.info("Oppdaterer NavAnsatt med dato for sletting azureId=${ansatt.azureId} dato=$deletionDate")
             val ansattToDelete = ansatt.copy(roller = emptySet(), skalSlettesDato = deletionDate)
-            ansatte.upsert(NavAnsattDbo.fromNavAnsattDto(ansattToDelete))
+            navAnsattRepository.upsert(NavAnsattDbo.fromNavAnsattDto(ansattToDelete))
         }
 
-        val ansatteToDelete = ansatte.getAll(skalSlettesDatoLte = today)
+        val ansatteToDelete = navAnsattRepository.getAll(skalSlettesDatoLte = today)
         ansatteToDelete.forEach { ansatt ->
             logger.info("Sletter NavAnsatt fordi vi har passert dato for sletting azureId=${ansatt.azureId} dato=${ansatt.skalSlettesDato}")
-            db.transactionSuspend { tx ->
-                ansatte.deleteByAzureId(ansatt.azureId, tx)
-                deleteSanityAnsatt(ansatt)
-            }
+            deleteNavAnsatt(ansatt)
         }
     }
 
+    private suspend fun deleteNavAnsatt(ansatt: NavAnsattDto) {
+        val avtaleIds = avtaleRepository.getAvtaleIdsByAdministrator(ansatt.navIdent)
+
+        logger.info("Sletter NavAnsatt fordi vi har passert dato for sletting azureId=${ansatt.azureId} dato=${ansatt.skalSlettesDato}")
+        db.transactionSuspend { tx ->
+            navAnsattRepository.deleteByAzureId(ansatt.azureId, tx)
+            deleteSanityAnsatt(ansatt)
+        }
+
+        avtaleIds
+            .forEach {
+                val avtale = avtaleRepository.get(it) ?: throw Exception("")
+                if (avtale.administratorer.isEmpty()) {
+                    notifyRelevantAdministrators(avtale, ansatt.hovedenhet)
+                }
+            }
+    }
+
+    private fun notifyRelevantAdministrators(
```

Comment:
Denne er da basert på forrige administrators fylke, i motsetning til å bruke arenaAnsvarligEnhet eller kontorstruktur på avtale. Litt usikker på hva som er best men tenkte det ga mening.

Reply:
Synes egentlig det gir mening å benytte avtalens fylke, men kanskje helst fra `avtale_nav_enhet`-relasjonen? :thinking: 

Koden er forøvrig veldig lik det som finnes arenaadapter-service, er det noe tanke om å generalisere, eller heller ta det om dette notifikasjonsscenarioet skulle oppstå enda en gang?